src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -649,8 +649,14 @@ public void SetLastCompletedClusterTransactionIndex(DocumentsOperationContext co
                 tree.Add(LastCompletedClusterTransactionIndexSlice, indexSlice);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public IEnumerable<Document> GetDocumentsStartingWith(DocumentsOperationContext context, string idPrefix, string startAfterId,
             long start, long take, string collection, Reference<long> skippedResults, DocumentFields fields = DocumentFields.All, CancellationToken token = default)
+            => GetDocumentsStartingWith(context, idPrefix, startAfterId, start, take, collection, skippedResults, fields, token);
```

Comment:
```suggestion
            => GetDocumentsStartingWith<Document>(context, idPrefix, startAfterId, start, take, collection, skippedResults, fields, token);
```

otherwise it's recursive call

Reply:
BTW, we need to think about where this scenario might happen and add a test for it.

Reply:
It was unused, I've removed it. The only call to `GetDocumentsStartingWith` is from `CollectionQuery` which is now changed to `QueriedDocument`.